### PR TITLE
ZENKO-2833-testing-list-buckets-components-and-actions

### PR DESCRIPTION
**Pull request by romainlancia**

### CODE REVIEW

File: src/react/actions/__tests__/s3bucket.test.jsx [hunk 1]
```diff
@@ -43,6 +53,16 @@ describe('s3bucket actions', () => {
                 dispatchAction.NETWORK_END_ACTION,
             ],
         },
+        {
+            it: 'listBuckets: should return list of buckets',
```
Comment (nicolas2bert) [inline src/react/actions/__tests__/s3bucket.test.jsx:68]: "Should handle error" test?

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        const bucketRows = component.find('Row');
+        bucketRows.forEach((bucketRow, index) => {
+            if (buckets[index].Name === 'bucket1') {
+                expect(bucketRow.find('tr').parent().prop('isSelected')).toBe(false);
+            } else if (buckets[index].Name === 'bucket2') {
+                expect(bucketRow.find('tr').parent().prop('isSelected')).toBe(true);
+            }
+        });
+
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: NIT: extra line.

File: src/react/actions/__tests__/s3bucket.test.jsx [hunk 3]
```diff
@@ -2,13 +2,23 @@ import * as actions from '../s3bucket';
...
+    errorS3State, initState, testActionFunction, testDispatchFunction,
 } from './utils/testUtil';
 
 const createBucketNetworkAction = dispatchAction.NETWORK_START_ACTION('Creating bucket');
 const listBucketsNetworkAction = dispatchAction.NETWORK_START_ACTION('Listing buckets');
 
 describe('s3bucket actions', () => {
+    const syncTests = [
+        {
+            it: 'should return OPEN_ACCOUNT_DELETE_DIALOG action',
```
Comment (nicolas2bert) [inline src/react/actions/__tests__/s3bucket.test.jsx]: Incorrect description.

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import BucketList from '../BucketList';
+import { MemoryRouter } from 'react-router-dom';
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('BucketList', () => {
+    const buckets = [{
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: buckets should be an immutable List.

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    });
+
+    it('should select row if the bucket name specified in the parameter matches one of the bucket names listed', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName="bucket2"/>
+            </MemoryRouter>,
+        );
+
+        const bucketRows = component.find('Row');
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: If you import and find the `Row` components directly, you won't have to  `.find('tr').parent()`.

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        bucketRows.forEach((bucketRow, index) => {
+            if (buckets[index].Name === 'bucket1') {
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: We don't know that `bucketRows` and `buckets` arrays will stay sorted in the same way. 
If we decide to sort our `<Row>`s by bucket name for example, code will still be correct (meaning that the right `<Row>` will be selected) but the tests will fail.

File: src/react/databrowser/buckets/__tests__/Buckets.test.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        CreationDate: 'Wed Oct 07 2020 16:35:57',
+        LocationConstraint: 'us-east-1',
+        Name: 'bucket1',
+    }, {
+        CreationDate: 'Wed Oct 07 2020 16:35:57',
+        LocationConstraint: 'us-east-1',
+        Name: 'bucket2',
+    }];
+
+    it('should display EmptyStateContainer if no buckets is present', () => {
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/Buckets.test.jsx]: NIT: no bucket

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        expect(secondBucketCellLink.text()).toBe('bucket2');
+        expect(secondBucketCellLink.prop('href')).toBe('/buckets/bucket2/objects');
+        expect(secondBucketCellLocation.text()).toBe('us-east-1 / Zenko Local Filesystem');
+        expect(secondBucketCellDate.text()).toBe('Wed Oct 07 2020 16:35:57');
+    });
+
+    it('should select row if the bucket name specified in the parameter matches one of the bucket names listed', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName="bucket2"/>
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: `locations` can not be of type string. It is an "object as map". Its zero-value can be set to `{}`

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                expect(bucketRow.find('tr').parent().prop('isSelected')).toBe(true);
+            }
+        });
+
+    });
+
+    it('should select no row if there is no bucket name specified in the parameter', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName=""/>
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: Same comment it for `locations`.

**Review by nicolas2bert — CHANGES_REQUESTED** [review]

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 9 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 8 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/Buckets.test.jsx (hunk 7 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/Buckets.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 4 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/actions/__tests__/s3bucket.test.jsx (hunk 3 above)
Comment (romainlancia) [reply src/react/actions/__tests__/s3bucket.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 2 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 6 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/databrowser/buckets/__tests__/BucketList.test.jsx (hunk 5 above)
Comment (romainlancia) [reply src/react/databrowser/buckets/__tests__/BucketList.test.jsx]: done

File: src/react/actions/__tests__/s3bucket.test.jsx (hunk 1 above)
Comment (romainlancia) [reply src/react/actions/__tests__/s3bucket.test.jsx:68]: done

**Comment by nicolas2bert** [comment]
You need your JIRA ticket number in your commit.

**Comment by romainlancia** [comment]
I just reword the commit message with the ticket number inside

**Review by nicolas2bert — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Comment by romainlancia** [comment]
@bert-e approve

**Pull request merged**